src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,138 @@
+                    for (int i = 0; i < items.size(); i++) {
+                        messages.add(i + 1 + "." + items.get(i));
+                    }
+                    printWithLine(messages);
+
+                } else if (line.split(" ")[0].equalsIgnoreCase("done")) {
+                    printWithLine(List.of());
+                    Task markItem = items.get(Integer.parseInt(line.substring(5)) - 1);
+                    markItem.markAsDone();
+                    printWithLine(List.of("Nice! I've marked this task as done: ", " " + markItem));
+
+                } else if (line.split(" ")[0].equalsIgnoreCase("todo")) {
+                    printWithLine(List.of());
+                    Task todoTask = new Todo(line.replace(line.split(" ")[0] + " ", ""));
+                    items.add(todoTask);
+                    printWithLine((List.of("Got it. I've added this task: ", todoTask.toString(), "Now you have " + items.size() + " task in the list. ")));
+
+                } else if (line.split(" ")[0].equalsIgnoreCase("deadline")) {
+                    printWithLine(List.of());
+                    int position = line.indexOf("/");
+                    String time = line.split("/")[1].replace("by ", "");
+                    Task deadlineTask = new Deadline(line.substring(9, position - 1), time);
+                    items.add(deadlineTask);
+                    printWithLine((List.of("Got it. I've added this task: ", deadlineTask.toString(), "Now you have " + items.size() + " task in the list. ")));
+
+                } else if (line.split(" ")[0].equalsIgnoreCase("event")) {
+                    printWithLine(List.of());
+                    int position = line.indexOf("/");
+                    String time = line.split("/")[1].replace("at ", "");
+                    Task eventTask = new Event(line.substring(6, position - 1), time);
+                    items.add(eventTask);
+                    printWithLine((List.of("Got it. I've added this task: ", eventTask.toString(), "Now you have " + items.size() + " task in the list. ")));
+
+                } else if (line.equalsIgnoreCase("bye")) {
+                    printWithLine(List.of());
+                    printWithLine(List.of("Bye. Hope to see you again soon!"));
+
+                }
+            }
+            catch (DukeCheckLineException e){
+                printWithLine(List.of());
+                printWithLine(List.of("☹ OOPS!!! I'm sorry, but I don't know what that means :-("));
+            }
+            catch (DukeCheckLineEmptyException e){
+                printWithLine(List.of());
+                printWithLine(List.of("☹ OOPS!!! Please enter somethings."));
+            }
+            catch (DukeException e){
+                printWithLine(List.of());
+                printWithLine((List.of("☹ OOPS!!! The description of a " + keyword + " cannot be empty.")));
+            }
+
+        }while(!line.equals("bye"));
+    }
```

Comment:
Perhaps you can try to break the long method into many short methods to improve the readability.

Reply:
Ok, thanks for your advice.